python/examples/min_distance.py

```diff
@@ -111,25 +112,30 @@ def argmin_1d(float(N) S, int32(D) MinIdx) -> (min_idx) {
 codes_t = torch.from_numpy(codes).cuda()
 luts_t = torch.from_numpy(luts).cuda()
 
-reduce_codes = tc.define(lang, name="reduce_codes")
-min_2d = tc.define(lang, name="min_2d")
-min_1d = tc.define(lang, name="min_1d")
-argmin_2d = tc.define(lang, name="argmin_2d")
-argmin_1d = tc.define(lang, name="argmin_1d")
-
-S = reduce_codes(luts_t, codes_t)
-V = min_2d(S.view((D, N / D)))
-v = min_1d(V)
-MinIdx = argmin_2d(S.view((D, N / D)), v)
-min_idx = argmin_1d(S, MinIdx)
+reduce_codes = tc.define(
+    lang, entry_point="reduce_codes", fallback=tc.MappingOptions('naive'))
+min_2d = tc.define(
+    lang, entry_point="min_2d", fallback=tc.MappingOptions('naive'))
+min_1d = tc.define(
+    lang, entry_point="min_1d", fallback=tc.MappingOptions('naive'))
+argmin_2d = tc.define(
+    lang, entry_point="argmin_2d", fallback=tc.MappingOptions('naive'))
+argmin_1d = tc.define(
+    lang, entry_point="argmin_1d", fallback=tc.MappingOptions('naive'))
+
+S, = reduce_codes(luts_t, codes_t)
+V, = min_2d(S.view((D, N / D)))
+v, = min_1d(V)
+MinIdx, = argmin_2d(S.view((D, N / D)), v)
```

Comment:
I have used lower case v because it is a value, single capital letter for parameters, CamelCase for tensor names and snake_case is for tc names as python objects.

I see there is a potential collision for V which looks like a parameter but is a tensor.
How about v, V -> minValue, MinTensor